types/src/database_sql.rs

```diff
@@ -46,6 +47,75 @@ impl Database {
         Ok(Database { client: conn })
     }
 
+    /// Get a specific comment by id
+    pub async fn comment_by_id(&self, comment_id: &Uuid) -> Result<Comment> {
+        Ok(
+            query_file_as!(BasicComment, "queries/comment_by_id.sql", comment_id,)
+                .fetch_one(&self.client)
+                .await?
+                .into(),
+        )
+    }
+
+    /// Get all comments on a given object
+    pub async fn comments_by_parent(
+        &self,
+        parent_id: &Uuid,
+        parent_type: &CommentParentType,
+    ) -> Result<Vec<Comment>> {
+        Ok(query_file_as!(
+            BasicComment,
+            "queries/comments_by_parent.sql",
+            parent_id,
+            parent_type.clone() as CommentParentType
+        )
+        .fetch_all(&self.client)
+        .await?
+        .into_iter()
+        .map(|c| c.into())
```

Comment:
There may be syntactic sugar for this (Like `Into::into` or something) but I couldn't get it to work.

Reply:
I think this is perfectly fine tbh, it feels clear on first glance so I dont think any syntactic sugar would be better